pkg/resources/scim_integration.go

```diff
@@ -74,140 +86,170 @@ func SCIMIntegration() *schema.Resource {
 	}
 }
 
-// CreateSCIMIntegration implements schema.CreateFunc.
-func CreateSCIMIntegration(d *schema.ResourceData, meta interface{}) error {
+// CreateContextSCIMIntegration implements schema.CreateFunc.
+func CreateContextSCIMIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
 	name := d.Get("name").(string)
+	id := sdk.NewAccountObjectIdentifier(name)
+	enabled := d.Get("enabled").(bool)
+	scimClient := d.Get("scim_client").(string)
+	runAsRole := d.Get("run_as_role").(string)
 
-	stmt := snowflake.NewSCIMIntegrationBuilder(name).Create()
-
-	// Set required fields
-	stmt.SetRaw(`TYPE=SCIM`)
-	stmt.SetString(`SCIM_CLIENT`, d.Get("scim_client").(string))
-	stmt.SetString(`RUN_AS_ROLE`, d.Get("provisioner_role").(string))
+	req := sdk.NewCreateScimSecurityIntegrationRequest(id, enabled, sdk.ScimSecurityIntegrationScimClientOption(scimClient), sdk.ScimSecurityIntegrationRunAsRoleOption(runAsRole))
 
-	// Set optional fields
-	if _, ok := d.GetOk("network_policy"); ok {
-		stmt.SetString(`NETWORK_POLICY`, d.Get("network_policy").(string))
+	// Set optionals
+	if v, ok := d.GetOk("network_policy"); ok {
+		req.WithNetworkPolicy(sdk.NewAccountObjectIdentifier(v.(string)))
 	}
-
-	if err := snowflake.Exec(db, stmt.Statement()); err != nil {
-		return fmt.Errorf("error creating security integration")
+	req.WithSyncPassword(d.Get("sync_password").(bool))
+	if v, ok := d.GetOk("comment"); ok {
+		req.WithComment(v.(string))
+	}
+	if err := client.SecurityIntegrations.CreateScim(ctx, req); err != nil {
+		return diag.FromErr(err)
 	}
 
-	d.SetId(name)
+	d.SetId(helpers.EncodeSnowflakeID(id))
 
-	return ReadSCIMIntegration(d, meta)
+	return ReadContextSCIMIntegration(ctx, d, meta)
 }
 
-// ReadSCIMIntegration implements schema.ReadFunc.
-func ReadSCIMIntegration(d *schema.ResourceData, meta interface{}) error {
+// ReadContextSCIMIntegration implements schema.ReadFunc.
+func ReadContextSCIMIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
-	id := d.Id()
-
-	stmt := snowflake.NewSCIMIntegrationBuilder(id).Show()
-	row := snowflake.QueryRow(db, stmt)
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
 
-	// Some properties can come from the SHOW INTEGRATION call
-
-	s, err := snowflake.ScanScimIntegration(row)
+	integration, err := client.SecurityIntegrations.ShowByID(ctx, id)
 	if err != nil {
-		return fmt.Errorf("could not show security integration")
+		log.Printf("[DEBUG] notification integration (%s) not found", d.Id())
```

Comment:
Check for `errors.Is(sdk.ErrObjectNotFound, err)` and only then remove the resources `d.SetId("")`

Reply:
Done.